recipes/iceoryx/all/conanfile.py

```diff
@@ -0,0 +1,254 @@
+        "shared":           False,
+        "toml_config":      True
+    }
+    generators = ["cmake", "cmake_find_package"]
+    exports_sources = ["patches/**"]
+    cmake = None
+
+    @staticmethod
+    def _create_cmake_module_alias_target(module_file, alias, aliased):
+        content = ""
+        content += textwrap.dedent("""\
+            if(TARGET {aliased} AND NOT TARGET {alias})
+                add_library({alias} INTERFACE IMPORTED)
+                set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+            endif()
+        """.format(alias=alias, aliased=aliased))
+        tools.save(module_file, content)
+
+    @property
+    def _module_subfolder(self):
+        return os.path.join(
+            "lib",
+            "cmake"
+        )
+
+    @property
+    def _conan_cmake_file(self):
+        return os.path.join(
+            os.getcwd(),
+            "conanbuildinfo.cmake"
+        )
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_folder(self):
+        return "build"
+
+    @property
+    def _pkg_share(self):
+        return os.path.join(
+            self.package_folder,
+            "share"
+        )
+
+    @property
+    def _pkg_etc(self):
+        return os.path.join(
+            self.package_folder,
+            "etc"
+        )
+
+    @property
+    def _pkg_bin(self):
+        return os.path.join(
+            self.package_folder,
+            "bin"
+        )
+
+    @property
+    def _pkg_cmake(self):
+        return os.path.join(
+            self.package_folder,
+            "lib/cmake"
+        )
+
+    @property
+    def _cmake_folder(self):
+        return os.path.join(
+            self._source_subfolder,
+            "iceoryx_meta"
+        )
+
+    @property
+    def _target_aliases(self):
+        return {
+            "iceoryx_posh::iceoryx_posh": "iceoryx::posh",
+            "iceoryx_posh::iceoryx_posh_roudi": "iceoryx::posh_roudi",
+            "iceoryx_binding_c::iceoryx_binding_c": "iceoryx::binding_c",
+            "iceoryx_utils::iceoryx_utils": "iceoryx::utils"
+        }
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "14")
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Windows currently not supported")
+
+    def config_options(self):
+        if self.options.toml_config:
+            self.requires("cpptoml/0.1.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True,
+                  destination=self._source_subfolder)
+
+    def build(self):
+        self._patch_sources()
+
+        self.cmake = CMake(self)
+        self.cmake.definitions["BUILD_SHARED_LIBS"]     = self.options.shared
```

Comment:
```suggestion
```

CMake helper does it, it's magic: https://docs.conan.io/en/latest/reference/build_helpers/cmake.html?highlight=build_shared_libs#definitions